kong/plugins/http-log/handler.lua

```diff
@@ -76,18 +79,25 @@ local function send_payload(self, conf, payload)
     end
   end
 
+  table_clear(headers_cache)
```

Comment:
@dndx I don't think this is correct.

Below the `httpc` makes a request, which involves connecting and sending socket operations. If that blocks for a while waiting for sockets to become available, then a follow up request might clear the table, before it was send. and hence break the whole thing. Nasty race condition.

Reply:
That's a good observation. However, `resty.http` always copies the params table when calling `:request`:

https://github.com/ledgetech/lua-resty-http/blob/master/lib/resty/http.lua#L781

Because later it needs to change the meltable on the `params` table https://github.com/ledgetech/lua-resty-http/blob/master/lib/resty/http.lua#L605 so it is not a good idea to reuse it.

Reply:
Slightly related: https://github.com/ledgetech/lua-resty-http/issues/103